auction-app-backend/src/main/java/com/example/auctionapp/util/csv/CsvUtil.java

```diff
@@ -0,0 +1,116 @@
+package com.example.auctionapp.util.csv;
+
+import com.example.auctionapp.entity.CategoryEntity;
+import com.example.auctionapp.entity.CreditCardEntity;
+import com.example.auctionapp.entity.PaymentInfoEntity;
+import com.example.auctionapp.entity.ProductEntity;
+import com.example.auctionapp.entity.ProductImageEntity;
+import com.example.auctionapp.entity.UserEntity;
+import com.example.auctionapp.entity.enums.ProductStatus;
+import com.example.auctionapp.exceptions.repository.ResourceNotFoundException;
+import com.example.auctionapp.repository.CategoryRepository;
+import com.example.auctionapp.repository.ProductImageRepository;
+import com.example.auctionapp.repository.ProductRepository;
+import com.example.auctionapp.util.builderpattern.GenericBuilder;
+import com.opencsv.bean.CsvToBean;
+import com.opencsv.bean.CsvToBeanBuilder;
+import com.opencsv.bean.HeaderColumnNameMappingStrategy;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class CsvUtil {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    @Transactional
+    public List<ProductEntity> uploadProduct(final MultipartFile file,
```

Comment:
This is a util file 
Purpose of this file should be to accept the data and process them in a way that fits our need (your private mehtod bellow)
It should save anything to the repository, for that we have a service 
So flow should be 
Controller accepts the request and pass it to the service 
Service pass to the csv util file request and here we format data in a way that fits us 
CSV util process the data and returns them in a correct format to the service
Service takes the data and uses the repository interface to store the data

Reply:
i fixed it by moving the save all method to product service, i also removed product image repository from the util as well. i left category repository in the util, so i can check if category exists for the sent category name. should i move the category repository as well or is it fine to leave it there since it does not store or modify the data?